File: Week 08/id_454/LeetCode_300_454.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function lengthOfLIS($nums)
    {
        if (count($nums) == 0) return 0;
        $dp = array_fill(0, count($nums), 1);
        for ($i = 0; $i < count($nums); $i++) {
            for ($j = 0; $j < $i; $j++) {
                if ($nums[$i] > $nums[$j])
                    $dp[$i] = max($dp[$i], $dp[$j] + 1);
            }
        }
        return max($dp);
    }
}

File: Week 08/id_454/LeetCode_300_454.go
func lengthOfLIS(nums []int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}
	dp := make([]int, len(nums))
	for i := range dp {
		dp[i] = 1
	}

	result := 1
	for i := 1; i < len(nums); i++ {
		for j := 0; j < i; j++ {
			if nums[i] > nums[j] {
				dp[i] = max(dp[j]+1, dp[i])
			}
		}
		if dp[i] > result {
			result = dp[i]
		}
	}

	return result
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}